pkg/operator/ceph/cluster/status/conditions.go

```diff
@@ -0,0 +1,300 @@
+/*
+Copyright 2016 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package status to provide conditions for CephCluster
+package status
+
+import (
+	"time"
+
+	"github.com/coreos/pkg/capnslog"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	logger    = capnslog.NewPackageLogger("github.com/rook/rook", "op-status")
+	namespace string
+	name      string
+)
+
+func ClusterInfo(Namespace, Name string) {
+	namespace = Namespace
+	name = Name
+}
+
+// updates the conditions of the cluster custom resource
+func setCondition(context *clusterd.Context, conditions *[]cephv1.RookConditions, newCondition cephv1.RookConditions) {
+	if conditions == nil {
+		conditions = &[]cephv1.RookConditions{}
+	}
+	existingCondition := findStatusCondition(*conditions, newCondition.Type)
+	if existingCondition == nil {
+		newCondition.LastTransitionTime = metav1.NewTime(time.Now())
+		newCondition.LastHeartbeatTime = metav1.NewTime(time.Now())
+		applyCondition(context, conditions, newCondition)
+		return
+	}
+	if existingCondition.Status != newCondition.Status {
+		removeStatusCondition(conditions, newCondition.Type)
+		newCondition.LastTransitionTime = metav1.NewTime(time.Now())
+		applyCondition(context, conditions, newCondition)
+	}
+}
+
+func ConditionConnecting(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
```

Comment:
Put a comment before explaining what that function does.

Reply:
Yes.